lib/ansible/modules/cloud/docker/docker_login.py

```diff
@@ -49,7 +49,7 @@
   email:
     required: False
     description:
-      - "The email address for the registry account."
+      - Does nothing, do not use.
```

Comment:
This sounds a bit like backwards compatibility is not given. Can you explain why it is no longer needed?

Right now (before this PR), the email address is used for authentication with docker registries, in particular Docker Hub (or private registries protected with HTTP Basic Auth). How can the module authenticate to Docker Hub without one half of the login credentials?

Reply:
As far as I am aware, email is not used for authentication. In fact the present [docker login ...](https://docs.docker.com/engine/reference/commandline/login/) command doesn't give you a way to set it at all. The user name and password are joined with a `:` and then base64 encoded. It does appear that `email` can be [sent in the request](https://docs.docker.com/engine/api/v1.40/#section/Authentication) thought. Still, the [credentials store api](https://github.com/docker/docker-py/blob/master/docker/credentials/store.py#L49-L58) has no way to store an email value. So I'm not really sure what to do with it ...

Reply:
Ah, you're right, the email isn't sent. I misremembered that apparently.

I guess Docker originally thought it might be used, but then dropped that... Then it should be OK to remove it.